### ISSUE TITLE
[BUG] Memgraph gets stuck when creating nodes in parallel with indexes

### ISSUE THREAD

**Issue by alex-linx** [memgraph/memgraph#1165]
**Memgraph version**
Memgraph v 2.10

**Environment**
Running on memgraph cloud, US East, 16GB instance.
Connecting using mgclient + memgraph lab web interface.

**Describe the bug**
If I have multiple queries running in parallel, that are creating nodes that have an index on them,
the db crashes. In the cloud interface it says running, but I can't connect in any way.

The DB seems to stay that way for hours, the only way I found to restore it is by pausing and resuming the project.

**To Reproduce**
The following python code reproduces the bug.
It has 100 jobs, each job creates an index on label `TestNode{k}` for some random `k`,
and then executes a query to create 10K nodes of such label.
After a few seconds the db crashes (although there is no indication for it).

EDIT:
~If I remove the creation of indexes - then the whole 1M nodes get's inserted in a few seconds without issue~
It seems that maybe the indexes are not the issue but the number of parallel jobs. Not sure...

EDIT 2:
Another possibility is what's causing it is the `STORAGE MODE IN_MEMORY_ANALYTICAL`.

```
def run_parallel():
    MEMGRAPH_HOST = os.getenv("MEMGRAPH_HOST")
    MEMGRAPH_USERNAME = os.getenv("MEMGRAPH_USERNAME")
    MEMGRAPH_PASSWORD = os.getenv("MEMGRAPH_PASSWORD")

    conn = mgclient.connect(
        host=MEMGRAPH_HOST,
        username=MEMGRAPH_USERNAME,
        password=MEMGRAPH_PASSWORD,
        sslmode=mgclient.MG_SSLMODE_REQUIRE,
        port=7687,
    )
    conn.autocommit = True
    cursor = conn.cursor()
    cursor.execute("STORAGE MODE IN_MEMORY_ANALYTICAL")
    conn.close()

    def run_queries():
        conn = mgclient.connect(
            host=MEMGRAPH_HOST,
            username=MEMGRAPH_USERNAME,
            password=MEMGRAPH_PASSWORD,
            sslmode=mgclient.MG_SSLMODE_REQUIRE,
            port=7687,
        )
        conn.autocommit = True
        cursor = conn.cursor()
        N = 10_000
        label_index = random.randint(0, 20)
        label = f"TestNode{label_index}"
        queries = [f"CREATE (:{label} {{index: {i}}})" for i in range(N)]
        print("Running queries")
        cursor.execute(f"CREATE INDEX ON :{label}")  # <-- Removing this line seems to make the issue disappear
        cursor.execute(" ".join(queries))
        conn.close()

    n_jobs = 100
    Parallel(n_jobs=n_jobs, max_nbytes=None)(delayed(run_queries)() for _ in range(n_jobs))
```

EDIT 3:
OK I think it definitely does have something to do with the indexes.
If I create the indexes before creating the nodes - the process finishes in about 1-2 mins for 1M nodes.
My theory is what happens that when there are 100 jobs, and each chooses a label at random,
so maybe it tries to create an index on the same label concurrently and that's what causing the issue.

**Expected behavior**
I would expect the db not to crash, or at least have some indication as to what's going on.

**Logs**
If applicable, add logs of Memgraph, CLI output or screenshots to help explain
your problem.

**Comment by gitbuda** [comment]
@alex-linx, thanks for reporting! This seems like a bigger issue, we'll try to investigate ASAP

**Comment by gvolfing** [comment]
@alex-linx
Memgraph seems to be deadlocking when running this script, and sometimes segfaults happen as well. We are investigating this issue.
However if this is something you would like to use in production, the usual use-case for indices is that they are created first, not on the fly. If you create the indices first and start creating the nodes concurrently, this issue does not persist.
But again, even with on-the-fly index creation, this should work and we are looking into this.

**Comment by gitbuda** [comment]
@alex-linx this feels like an amazing stress test for index creation + adding data concurrently -> `memgraph` is going to improve 💪 

But do you have that type of workload in the production? In general, indexes are created once (at any point during the database workload). **Could you elaborate what's the actual production workload on your end?**

**Comment by antejavor** [comment]
We have one more report. On exactly this issue, but the Memgraph is IN_MEMORY_TRANSACTIONAL mode.

Additional context: 
https://discord.com/channels/842007348272169002/890969266600423504/1145981066075701299

**Comment by katarinasupe** [comment]
@gitbuda @gvolfing, what is the update on this issue since it's blocking community users in production?

**Comment by gvolfing** [comment]
@katarinasupe this is the next priority.

**Comment by gitbuda** [comment]
Closing this issue because it seems there is a workaround + we have another issue created which should a the proper improvement but it's more complex
